src/main/java/htsjdk/samtools/cram/io/DefaultBitInputStream.java

```diff
@@ -89,49 +82,16 @@ private static int rightBits(final int n, final int x) {
         return x & ((1 << n) - 1);
     }
 
-    @Override
-    public final long readLongBits(int n) {
-        if (n > 64)
-            throw new RuntimeException("More then 64 bits are requested in one read from bit stream.");
-
-        if (n == 0)
-            return 0;
-
-        long x = 0;
-        long byteBuffer = this.byteBuffer;
-        if (nofBufferedBits == 0) {
-            try {
-                byteBuffer = in.read();
-            } catch (final IOException e) {
-                throw new RuntimeIOException(e);
-            }
-            if (byteBuffer == -1) {
-                throw new RuntimeEOFException("End of stream.");
-            }
-            nofBufferedBits = 8;
-        }
-        byteBuffer &= masks[nofBufferedBits];
-        while (n > nofBufferedBits) {
-            n -= nofBufferedBits;
-            x |= byteBuffer << n;
-            try {
-                byteBuffer = in.read();
-            } catch (final IOException e) {
-                throw new RuntimeIOException(e);
-            }
-            if (byteBuffer == -1) {
-                throw new RuntimeEOFException("End of stream.");
-            }
-            nofBufferedBits = 8;
-        }
-        nofBufferedBits -= n;
-        this.byteBuffer = (int) (byteBuffer & masks[nofBufferedBits]);
-        return x | (byteBuffer >>> nofBufferedBits);
-    }
-
     @Override
     public void reset() {
         nofBufferedBits = 0;
         byteBuffer = 0;
+
+        try {
+            super.reset();
```

Comment:
This was a bug.  It didn't reset() the underlying stream.

Reply:
Found by a new test in this PR

Reply:
Nasty bug